src/ComputeSharp.SourceGeneration/Models/TypeInfo.cs

```diff
@@ -13,6 +13,21 @@ namespace ComputeSharp.SourceGeneration.Models;
 /// <param name="IsRecord">Whether the type is a record type.</param>
 internal sealed record TypeInfo(string QualifiedName, TypeKind Kind, bool IsRecord)
 {
+    /// <summary>
+    /// Gets the keyword for the current type kind.
+    /// </summary>
+    /// <returns>The keyword for the current type kind.</returns>
+    public string GetTypeKeyword()
+    {
+        return Kind switch
+        {
+            TypeKind.Struct => "struct",
+            TypeKind.Interface => "interface",
+            TypeKind.Class when IsRecord => "record",
+            _ => "class"
+        };
```

Comment:
Intentionally not handling record structs?

Reply:
Ooh good catch! I think I wrote that before record structs existed and forgot to add it 😄
Will fix this and add a new test to cover all possible scenarios in a separate PR.